pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -446,10 +485,19 @@ func (nc *Controller) doNoScheduleTaintingPassWorker() {
 			return
 		}
 		nodeName := obj.(string)
-
+		needsToRequeue := false
 		if err := nc.doNoScheduleTaintingPass(nodeName); err != nil {
```

Comment:
we need to check `TaintNodeByCondition` flag here, it's beta feature right now.

Reply:
Added in the new commit